### PR TITLE
Do not dump Babelfish initialize user

### PR DESCRIPTION
**Pull request by rishabhtanwar29**
### Description
When Babelfish initialize user is same between source and target servers
for dump/restore, the restore throws errors like "role ... already exists" and
"duplicate key value violates unique constraint..." etc while inserting catalog
entry for initialize user in babelfish_authid_login_ext catalog. These errors are
actually expected and makes sense as we know the user do already exists on
the target server but the whole restore gets rolled back in case the restore
process has been made transactional (`--single-transaction` option is used)
and there is no way for a user to fix this manually.

To overcome the above issue, the engine commit implements the logic to skip
dumping Babelfish initialize user as well as the "owner" column of
`sys.babelfish_sysdatabases` catalog table.

This commit implements logic to re-populate "owner" column of
`sys.babelfish_sysdatabases` catalog table during restore with the owner of
current database (which is going to be initialize user of target Babelfish database).

#### Test changes:
* Updated github dump/restore action to take two more options as input:
  1. dump-data-as: Dump table data using COPY or INSERT
  2. dump-format: Dump format (plain/custom/tar/directory)
* Updated dump-restore configurations to test all four dump formats and
  both data formats.
* Enabled `--single-transaction` option for restore so that any error during restore
  would result in restore getting rolledback and failed.
* Added TAP test coverage for pg_restore.
* Fixed BABEL-1206-vu-prepare test to insert valid data into varchar column. 

Task: BABEL-4404 
Signed-off-by: Rishabh Tanwar <ritanwar@amazon.com>

Engine PR: https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/pull/216

### Test Scenarios Covered ###
* **Use case based -**

* **Boundary conditions -**

* **Arbitrary inputs -**

* **Negative test cases -**

* **Minor version upgrade tests -**

* **Major version upgrade tests -**

* **Performance tests -**

* **Tooling impact -**

* **Client tests -**

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/composite-actions/dump-restore-util/action.yml [hunk 1]
```diff
@@ -63,35 +78,16 @@ runs:
...
-    - name: Temporarily disable failing tests
-      if: always() && steps.run-pg_dump-restore.outcome == 'success' && inputs.is_final_ver == 'true'
-      env:
-        migr_mode: ${{ inputs.migration_mode }}
-      run: |
-        if [[ "$migr_mode" == "multi-db" ]];then
-          base_dir=${{ matrix.upgrade-path.path[0] }}
-          if [[ "$base_dir" == *"latest"* ]]; then
-            base_dir="latest"
-          fi
```
Comment (HarshLunagariya) [inline .github/composite-actions/dump-restore-util/action.yml:84]: We don't need this because we modified testcase to use master, right?

File: contrib/babelfishpg_tsql/src/hooks.c [hunk 2]
```diff
@@ -3827,18 +3827,37 @@ fill_missing_values_in_copyfrom(Relation rel, Datum *values, bool *nulls)
...
+
+		attnum = (AttrNumber) attnameAttNum(rel, "owner", false);
+		Assert(attnum != InvalidAttrNumber);
 
-		dbid = getDbidForLogicalDbRestore(relid);
-		values[attnum - 1] = Int16GetDatum(dbid);
-		nulls[attnum - 1] = false;
+		if (nulls[attnum - 1])
+		{
+			const char *owner = GetUserNameFromId(GetSessionUserId(), false);
```
Comment (HarshLunagariya) [inline contrib/babelfishpg_tsql/src/hooks.c]: Will it be correct always? Or should we fetch current physical database's owner?

File: test/JDBC/input/pg_hint_plan/BABEL-3513-vu-verify.mix [hunk 3]
```diff
@@ -6,6 +6,9 @@ analyze master_dbo.babel_3513_t2;
 go
 
 -- tsql
+exec sp_babelfish_configure 'enable_pg_hint', 'on', 'server'
+go
```
Comment (HarshLunagariya) [inline test/JDBC/input/pg_hint_plan/BABEL-3513-vu-verify.mix:10]: Why this change?

File: .github/composite-actions/dump-restore-util/action.yml (hunk 1 above)
Comment (rishabhtanwar29) [reply .github/composite-actions/dump-restore-util/action.yml:84]: Yeah so, this step was initially added to temporarily skip the tests which were failing as part of dump/restore as we were still working on fixing them, but now this step is unnecessary since there are no test failures remaining and we do want to fail the workflow if there are any test failures.

File: test/JDBC/input/pg_hint_plan/BABEL-3513-vu-verify.mix (hunk 3 above)
Comment (rishabhtanwar29) [reply test/JDBC/input/pg_hint_plan/BABEL-3513-vu-verify.mix:10]: Oh this is needed because we don't preserve the database settings across dump/restore, so we basically need to reset the value of `enable_pg_hint`.

File: contrib/babelfishpg_tsql/src/hooks.c (hunk 2 above)
Comment (rishabhtanwar29) [reply contrib/babelfishpg_tsql/src/hooks.c]: Although both ways should be fine as restore must always be performed with babelfish initialize user but since we skipped dumping physical database's owner then it makes more sense to populate it here with physical database's owner here again.

File: test/JDBC/input/pg_hint_plan/BABEL-3513-vu-verify.mix (hunk 3 above)
Comment (HarshLunagariya) [reply test/JDBC/input/pg_hint_plan/BABEL-3513-vu-verify.mix:10]: Got it. But then won't we be missing coverage where it should be preserved  in case of M/mVU.

**Review by HarshLunagariya — APPROVED** [review]

**Review by shardgupta — APPROVED** [review]
